src/main/java/flight/reservation/plane/Helicopter.java

```diff
@@ -1,21 +1,45 @@
 package flight.reservation.plane;
 
 public class Helicopter extends Aircraft {
-    private final int H1_PASSENGER_CAPACITY = 4;
-    private final int H2_PASSENGER_CAPACITY = 6;
-    private final int CREW_PASSENGER_CAPACITY = 2;
-    private final String H1_MODEL_IDENTIFIER = "H1";
-    private final String H2_MODEL_IDENTIFIER = "H2";
+    private static final int H1_PASSENGER_CAPACITY = 4;
+    private static final int H2_PASSENGER_CAPACITY = 6;
+    private static final int CREW_PASSENGER_CAPACITY = 2;
+    private static final String H1_MODEL_IDENTIFIER = "H1";
+    private static final String H2_MODEL_IDENTIFIER = "H2";
+
+    private Helicopter() {
+    }
 
     public Helicopter(String model) {
-        setModel(model);
-        setCrewCapacity(CREW_PASSENGER_CAPACITY);
+        // We don't want to break the existing API (public constructor) here
+        // What would be a reasonable way to implement the factory method pattern then?
+        // Does it make sense at all?
+        Helicopter helicopter;
```

Comment:
Yes, when using a factory you cannot use the constructor any more so this would require a bigger rewrite.